backend/de.metas.business/src/main/java-legacy/org/compiere/model/MInvoice.java

```diff
@@ -1701,4 +1708,12 @@ private boolean annotateDocNoToDescription()
 		return annotateInvoice;
 	}
 
+	private BigDecimal apply5CentRoundingIfNeeded(@NonNull final BigDecimal grandTotal)
+	{
+
+		final IInvoiceBL invoiceBL = Services.get(IInvoiceBL.class);
+		return invoiceBL.roundTo5CentIfNeeded(InvoiceId.ofRepoId(getC_Invoice_ID()),
```

Comment:
you are already in MINvoice. Why do you have to send the invoiceId, then the BL will load gain and get the currency from it?
Why not send the invoice directly?

Reply:
or even better, send only `grandTotal`, `currencyId` and `SOTrx`.